gnocchi/tests/functional/gabbits/influxdb.yaml

```diff
@@ -0,0 +1,111 @@
+# Tests for the InfluxDB compatibility layer
+
+fixtures:
+    - ConfigFixture
+
+defaults:
+  request_headers:
+    # User admin
+    authorization: "basic YWRtaW46"
+    content-type: application/json
+
+tests:
+    - name: ping influxdb status with head
+      desc: test HEAD on ping – xfails because Pecan does not honor HEAD correctly yet
+      xfail: true
+      HEAD: /v1/influxdb/ping
+      status: 204
+
+    - name: ping influxdb status with get
+      GET: /v1/influxdb/ping
+      status: 204
+
+    - name: create a database
+      POST: /v1/influxdb/query?q=create+database+influxdbtest
+      status: 204
+
+    - name: check the resource type now exists
+      GET: /v1/resource_type/influxdbtest
+      status: 200
+      response_json_paths:
+        $:
+          name: influxdbtest
+          attributes: {}
+          state: active
+
+    - name: do an unrecognized query
+      POST: /v1/influxdb/query?q=select+metrics+plz
+      request_headers:
+        # This is useful to get the error in JSON format
+        accept: application/json
+      status: 501
+      response_json_paths:
+        $.description.cause: Not implemented error
+        $.description.detail: q
+        $.description.reason: Querying is not implemented
+
+    - name: create archive policy
+      POST: /v1/archive_policy
+      data:
+          name: low
+          definition:
+              - granularity: 1 hour
+      status: 201
+
+    - name: create archive policy for influxdb
+      POST: /v1/archive_policy_rule
+      data:
+        name: influxdb
+        metric_pattern: "*"
+        archive_policy_name: low
+      status: 201
+
+    - name: write a line
+      POST: /v1/influxdb/write?db=influxdbtest
+      request_headers:
+        content-type: text/plain
+      data:
+        "mymetric,host=foobar,mytag=myvalue field=123 1510581804179554816"
+      status: 204
+
+    - name: check resource created
+      GET: /v1/resource/influxdbtest/foobar
+      status: 200
+      response_json_paths:
+        $.original_resource_id: foobar
+        $.id: b4d568e4-7af1-5aec-ac3f-9c09fa3685a9
+        $.type: influxdbtest
+        $.creator: admin
+
+    - name: check metric created
+      GET: /v1/resource/influxdbtest/foobar/metric/mymetric.field@mytag=myvalue
```

Comment:
this should assert something.

Reply:
it does, it asserts there's no error being returned (i.e. no 404).

it does not check any content since it's not the purpose of this test. It's not testing the format, it's testing the metric creation.

Reply:
word. didn't realise gabbi would auto fail  for these cases. i'd prefer it be explicit and we validate more than just 'it didn't explode' but your call.

Reply:
yeah it's always a trade-off: asserting `200` for example would break if we ever change the code to be `202` for example, whereas here we don't really care.

Though I agree in this case there's 0% chance we ever change from 200 :)